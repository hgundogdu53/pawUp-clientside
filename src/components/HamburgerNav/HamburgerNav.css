br {
    display: block;
}
.menu-item {
    color: black;
    text-decoration: none;
}

img {
    width: 2rem;
    margin-top: 2.375rem;
}

img#paw-menu {
    width: 4rem;
}
#paw-menu {
    margin-top: 0;
  }

  .bm-item {
      display: inline-block;
      color: #d7efea;
    }
    
  .bm-item:hover {
      background-color: #004690;
      color: #d7efea;
      border-radius: 0.5rem;
  }