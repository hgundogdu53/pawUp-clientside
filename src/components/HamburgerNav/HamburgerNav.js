import React from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import LoggedInNav from '../LoggedInNav/LoggedInNav';
import { slide as Menu } from 'react-burger-menu';
import paws from '../../images/paws-logo.png';
import './HamburgerNav.css';

function HamburgerNav(props) {
    return (
        <>
            <Link to='/' className='nav-title'><img className='icon' src={paws} alt='paws-logo'></img><span className='title'>PawUp</span></Link>
            <>
                {/*Line 15-22 checks if the user is logged in */}
                {TokenService.hasAuthToken() ?
                    <LoggedInNav handleLogout={props.handleLogout} />
                    :
                    <Menu>
                        <Link to='/login' className="menu-item">Log In</Link><br />
                        <Link to='/signup' className="menu-item">Sign Up</Link>
                    </Menu>
                }
            </>
        </>
    );
}

export default HamburgerNav;
