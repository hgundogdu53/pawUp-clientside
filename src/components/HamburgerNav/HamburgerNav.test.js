import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HamburgerNav from './HamburgerNav';

describe(`HamburgerNav component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<HamburgerNav />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});