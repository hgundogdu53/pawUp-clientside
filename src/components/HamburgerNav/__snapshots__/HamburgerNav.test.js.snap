// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerNav component renders without crashing 1`] = `
<Fragment>
  <Link
    className="nav-title"
    to="/"
  >
    <img
      alt="paws-logo"
      className="icon"
      src="paws-logo.png"
    />
    <span
      className="title"
    >
      PawUp
    </span>
  </Link>
  <Menu
    bodyClassName=""
    burgerBarClassName=""
    burgerButtonClassName=""
    className=""
    crossButtonClassName=""
    crossClassName=""
    disableAutoFocus={false}
    disableCloseOnEsc={false}
    htmlClassName=""
    id=""
    itemClassName=""
    itemListClassName=""
    menuClassName=""
    morphShapeClassName=""
    noOverlay={false}
    noTransition={false}
    onStateChange={[Function]}
    outerContainerId=""
    overlayClassName=""
    pageWrapId=""
    right={true}
    styles={Object {}}
    width={300}
  >
    <Link
      className="menu-item"
      to="/login"
    >
      Log In
    </Link>
    <br />
    <Link
      className="menu-item"
      to="/signup"
    >
      Sign Up
    </Link>
  </Menu>
</Fragment>
`;
